src/EtherFiNode.sol

```diff
@@ -483,54 +485,55 @@ contract EtherFiNode is IEtherFiNode {
         return (payouts[0], payouts[1], payouts[2], payouts[3]);
     }
 
+    //--------------------------------------------------------------------------------------
+    //-------------------------------- CALL FORWARDING  ------------------------------------
+    //--------------------------------------------------------------------------------------
 
-    function callEigenPod(bytes memory data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
-        _verifyEigenPodCall(data);
-        return Address.functionCall(eigenPod, data);
+    function callEigenPod(bytes calldata _data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
+        _verifyEigenPodCall(_data);
+        return Address.functionCall(eigenPod, _data);
     }
 
-    // As an optimization, it skips the call to 'etherFiNodesManager' back again to retrieve the target address
-    function forwardCall(address to, bytes memory data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
-        _verifyForwardCall(to, data);
-        return Address.functionCall(to, data);
+    function forwardCall(address _to, bytes calldata _data) external onlyEtherFiNodeManagerContract returns (bytes memory) {
+        _verifyForwardCall(_to, _data);
+        return Address.functionCall(_to, _data);
     }
-    
-    //--------------------------------------------------------------------------------------
-    //-------------------------------  INTERNAL FUNCTIONS  ---------------------------------
-    //--------------------------------------------------------------------------------------
 
-    function _verifyEigenPodCall(bytes memory data) internal view {
-        bytes4 selector;
-        assembly {
-            selector := mload(add(data, 0x20))
-        }
+    function _verifyEigenPodCall(bytes calldata _data) internal view {
+
+        if (_data.length < 4) revert InvalidForwardedCall();
+        bytes4 selector = bytes4(_data[:4]);
+
+        // can add extra restrictions to specific calls here i.e. checking specific paramaters
+        if (!IEtherFiNodesManager(etherFiNodesManager).allowedForwardedEigenpodCalls(selector)) revert ForwardedCallNotAllowed();
```

Comment:
I think we can move all these checks on the forward calls to `EtherFiNodesManager` contract. just need to make suer the forward call are made via `EtherFiNodesManager` contract by putting `onlyEtherFiNodeManagerContract` where needed

Reply:
yep. I can do that. I assumed you wrote it like this originally due to size constraints of the manager contract so I left it. I will move it

Reply:
that is very true actually. how should we refactor EFNM to reduce its size :(

Reply:
my simplification work this week should in theory delete a lot of code out of EFNM. So I will go ahead and make the change in this PR and work to cut things out of EFNM in my next PR